mobile/src/reducers/user/user.action.ts

```diff
@@ -30,13 +30,44 @@ export interface LoadUserEmployeePermissions extends Action {
 
 export const loadUserEmployeePermissions = (employeeId: string): LoadUserEmployeePermissions => ({ type: 'LOAD-USER-EMPLOYEE-PERMISSIONS', employeeId });
 
-export interface LoadUserEmployeePermissionsFininshed extends Action {
+export interface LoadUserEmployeePermissionsFinished extends Action {
     type: 'LOAD-USER-EMPLOYEE-PERMISSIONS-FINISHED';
     permissions: UserEmployeePermissions;
 }
 
-export const loadUserEmployeePermissionsFinished = (permissions: UserEmployeePermissions): LoadUserEmployeePermissionsFininshed =>
+export const loadUserEmployeePermissionsFinished = (permissions: UserEmployeePermissions): LoadUserEmployeePermissionsFinished =>
     ({ type: 'LOAD-USER-EMPLOYEE-PERMISSIONS-FINISHED', permissions });
 
+export interface LoadUserPreferences extends Action {
+    type: 'LOAD-USER-PREFERENCES';
+    userId: string;
+}
+
+export const loadUserPreferences = (userId: string): LoadUserPreferences => ({ type: 'LOAD-USER-PREFERENCES', userId });
+
+export interface UpdateUserPreferences {
+    type: 'UPDATE-USER-PREFERENCES';
+    userId: string;
+    previousPreferences: UserPreferences;
+    preferences: UserPreferences;
+}
+
+export const updateUserPreferences = (userId: string, previousPreferences: UserPreferences, newPreferences: UserPreferences): UpdateUserPreferences => ({
+    type: 'UPDATE-USER-PREFERENCES',
+    userId: userId,
+    previousPreferences: previousPreferences,
+    preferences: newPreferences,
+});
+
+export interface LoadUserPreferencesFinished {
```

Comment:
Shouldn't it extends Action?

Reply:
6db94d99af30d768fd41fa902218d0b69551520f